### PR TITLE
uploadclient: upload from rsemgr - #3084

### PR DESCRIPTION
**Pull request by TomasJavurek**
uploadclient: upload from rsemgr - #3084

### CODE REVIEW

**Comment by TomasJavurek** [comment]
The version from a fresh new branch, because the previous one couldn't be easily rebased (too divereged). I tested this code by uploading a file to UNI-FRIEBURG under several conditions. All worked fine. This version is missing the attribute of the NoFilesUploaded exception, this would be done in separate ticket.

**Comment by bari12** [comment]
@mlassnig @TWAtGH please have a look

File: lib/rucio/client/uploadclient.py [hunk 1]
```diff
@@ -233,17 +237,18 @@ def upload(self, items, summary_file_path=None, traces_copy_out=None):
...
+                    logger.debug('Entering _upload_item')
+                    self._upload_item(rse_settings=rse_settings,
+                                      lfn=lfn,
+                                      source_dir=file['dirname'],
+                                      force_scheme=cur_scheme,
+                                      force_pfn=pfn,
+                                      transfer_timeout=file.get('transfer_timeout'),
+                                      delete_existing=delete_existing,
+                                      sign_service=sign_service)
+                    logger.debug('Physicaly uploaded.')
```
Comment (mlassnig) [inline lib/rucio/client/uploadclient.py]: Replace with simply "Upload done".

File: lib/rucio/client/uploadclient.py [hunk 2]
```diff
@@ -233,17 +237,18 @@ def upload(self, items, summary_file_path=None, traces_copy_out=None):
...
-                                          source_dir=file['dirname'],
-                                          force_scheme=cur_scheme,
-                                          force_pfn=pfn,
-                                          transfer_timeout=file.get('transfer_timeout'),
-                                          delete_existing=delete_existing,
-                                          sign_service=sign_service,
-                                          auth_token=self.auth_token)
-                    success = state['success']
-                    file['upload_result'] = state
+                    logger.debug('Entering _upload_item')
```
Comment (mlassnig) [inline lib/rucio/client/uploadclient.py]: I would remove this.

File: lib/rucio/client/uploadclient.py [hunk 3]
```diff
@@ -492,6 +497,150 @@ def _convert_file_for_api(self, file):
...
+                raise RucioException('Could not get protocol.stats for given PFN: %s' % pfn)
+
+            # The checksum and filesize check
+            if ('filesize' in stats) and ('filesize' in lfn):
+                if int(stats['filesize']) != int(lfn['filesize']):
+                    raise RucioException('Filesize after upload does not match the origin.')
+            if rse_settings['verify_checksum'] is not False:
+                if ('adler32' in stats) and ('adler32' in lfn):
+                    if stats['adler32'] != lfn['adler32']:
+                        raise RucioException('Checksum after upload does not match the origin.')
```
Comment (mlassnig) [inline lib/rucio/client/uploadclient.py]: Add both checksums  (source: %s, destination: %s) to error string.

File: lib/rucio/client/uploadclient.py [hunk 4]
```diff
@@ -492,6 +497,150 @@ def _convert_file_for_api(self, file):
...
+        # Checksum verification, obsolete, see Gabriele changes.
+        try:
+            stats = self._retry_protocol_stat(protocol_read, readpfn_tmp)
+            if not isinstance(stats, dict):
+                raise RucioException('Could not get protocol.stats for given PFN: %s' % pfn)
+
+            # The checksum and filesize check
+            if ('filesize' in stats) and ('filesize' in lfn):
+                if int(stats['filesize']) != int(lfn['filesize']):
+                    raise RucioException('Filesize after upload does not match the origin.')
```
Comment (mlassnig) [inline lib/rucio/client/uploadclient.py]: Add both filesizes (source: %s, destination: %s) to error string.

**Review by mlassnig — COMMENTED** [review]
few inline comments on the logging

**Pull request closed without merging**